### PR TITLE
Implement block generator:

### PR DESCRIPTION
**Pull request by grimadas**
## What is this pull request?
Genesis block generation added to iroha-cli.
   
## Why do you implement it? Why do we need this pull request?
Genesis block must be created for new iroha network with different peers. Peer address must be variable used in block generation.
  
## How to use the features provided in the pull request?
Run: 
`iroha-cli --genesis_block --peers_num 4 --peers_address ./peers.list`. 
Where peers.list is a file containing addresses (Ip or docker dns) of peers, e.g.: 
`
10.90.10.1:1001
10.90.10.2:1001
...
`

This command will create genesis.block in json format in the current directory. Genesis block can be used to start `irohad`

## Details/Features
List of features / major commits
- Add block, transaction, command generators
- Add signature generation with seed
- Refactor libs generator
- Remove unused imports
- Add to iroha-cli genesis block generation

### CODE REVIEW

**Comment by neewy** [comment]
Please, merge it to develop, until this is not a hotfix

**Comment by muratovv** [comment]
How pass public keys of peers to genesis block generator?

File: irohad/model/generators/impl/block_generator.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    namespace generators {
+      Block BlockGenerator::generateGenesisBlock(
+          std::vector<std::string> peers_address) {
+        Block block{};
+        block.created_ts =
+            (ts64_t)std::chrono::system_clock::now().time_since_epoch().count();
+        block.height = 1;
+        std::fill(block.prev_hash.begin(), block.prev_hash.end(), 0);
+        block.txs_number = 1;
+        block.sigs = {generateSignature(42)};
```
Comment (muratovv) [inline irohad/model/generators/impl/block_generator.cpp]: May be better to move out generation of signatures and pass it as a parameter. Better and more flexible interface is:
```C++
generateGenesisBlock(peers_address, generateSignature(42));
....
// in method
block.sigs = {signature_from_param}
```

File: irohad/model/generators/signature_generator.hpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#include "common/types.hpp"
+#include "generator/generator.hpp"
+#include "model/signature.hpp"
+
+namespace iroha {
+  namespace model {
+    namespace generators {
+
+      /**
+       *
```
Comment (muratovv) [inline irohad/model/generators/signature_generator.hpp]: Fill doc

**Comment by lebdron** [comment]
>`--peers_num 4 --peers_address ./peers.list`

Can we infer number of peers from list size?

**Pull request merged**